ocs_ci/ocs/amq.py

```diff
@@ -192,13 +247,227 @@ def setup_amq_kafka_bridge(self):
+        Return: kafka_topic object
+        """
+        try:
+            kafka_topic = templating.load_yaml(os.path.join(self.dir, self.kafka_topic_yaml))
+            # ToDo: To support multiple topics
+            kafka_topic["metadata"]["name"] = name
+            kafka_topic["spec"]["partitions"] = partitions
+            kafka_topic["spec"]["replicas"] = replicas
+            self.kafka_topic = OCS(**kafka_topic)
+            self.kafka_topic.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during creating of Kafka topic')
+            raise cf
+
+        # Making sure kafka topic created
+        if self.kafka_topic_obj.get(resource_name=name):
+            return self.kafka_topic
+        else:
+            raise ResourceWrongStatusException("kafka topic is not created")
+
+    def create_kafka_user(self, name="my-user"):
+        """
+        Creates kafka user
+
+        Args:
+             name (str): Name of the kafka user
+
+        Return: kafka_user object
+
+        """
+        try:
+            kafka_user = templating.load_yaml(os.path.join(self.dir, self.kafka_user_yaml))
+            # ToDo: Support multiple user
+            kafka_user["metadata"]["name"] = name
+            self.kafka_user = OCS(**kafka_user)
+            self.kafka_user.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during creating of Kafka user')
+            raise cf
+
+        # Making sure kafka user created
+        if self.kafka_user_obj.get(resource_name=name):
+            return self.kafka_user
+        else:
+            raise ResourceWrongStatusException("kafka user is not created")
+
+    def create_producer_pod(self, num_of_pods=1, value='10000'):
+        """
+        Creates producer pods
+
+        Args:
+            num_of_pods (int): Number of producer pods to be created
+            value (str): Number of the messages to be sent
+
+        Returns: producer pod object
+
         """
-        Setup AMQ from local folder,
-        function will call all necessary sub functions to make sure amq installation is complete
+        try:
+            producer_pod = templating.load_yaml(constants.HELLO_WORLD_PRODUCER_YAML)
+            producer_pod["spec"]["replicas"] = num_of_pods
+            producer_pod["spec"]["template"]["spec"]["containers"][0]["env"][4]["value"] = value
+            self.producer_pod = OCS(**producer_pod)
+            self.producer_pod.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during creation of producer pod')
+            raise cf
+
+        # Making sure the producer pod is running
+        if self.is_amq_pod_running(pod_pattern="hello-world-producer"):
+            return self.producer_pod
+        else:
+            raise ResourceWrongStatusException("producer pod is not getting to running state")
+
+    def create_consumer_pod(self, num_of_pods=1, value='10000'):
```

Comment:
This takes patch takes care of creating multiple producer and consumer pods or you have plans to implement in your next PR?

Reply:
By default, one producer and consumer pod created. We should be able to create multiple producer and consumer pod using this function, user can specify in param num_of_pods